### DISCUSSION TITLE
How to show percentage decimals of RAM ?

### DISCUSSION THREAD

**Discussion by Ayfri** [JanDeDobbeleer/oh-my-posh#1279] [Troubleshoot]
Hi, I'm trying to modify the template of the new `sysinfo` segment, but I can't figure out how to get the RAM showing percents decimals, even when putting 3 or 4 in the precision.

I have this segment : 
```json
{
  "type": "sysinfo",
  "style": "powerline",
  "invert_powerline": true,
  "powerline_symbol": "\uE0B2",
  "foreground": "#26C6DA",
  "background": "#2f2f2f",
  "properties": {
    "precision": 4,
    "prefix": " \uf85a ",
    "postfix": "% ",
    "template":"{{ .PhysicalPercentUsed }}"
  }
}
```

**Comment by JanDeDobbeleer** [comment]
@lnu do you think we can use sprig's [round](https://masterminds.github.io/sprig/mathf.html) function here in the template?

**Reply by lnu** [reply]
yes but by default it displays 2 digits precision:
```
        {
          "type": "sysinfo",
          "powerline_symbol": "",
          "foreground": "#ffffff",
          "background": "#8f43f3",
          "invert_powerline": true,
          "properties": {
            "prefix": " \uf85a ",
            "postfix": "% "
          },
          "style": "powerline"
        },
```

![image](https://user-images.githubusercontent.com/1829553/142845061-e7e36047-74c8-4111-bee4-f64e90a3160d.png)

or you can add this:
```
            "precision":1,
            "template": "{{ round .PhysicalPercentUsed .Precision }}"
```

**Reply by lnu** [reply]
and all info comes from here: https://psutil.readthedocs.io/en/latest/#psutil.virtual_memory

**Reply by Ayfri** [reply]
![image](https://user-images.githubusercontent.com/34159136/142846399-837da351-cfee-460d-ab40-4a226a8567be.png)
Not for me, I don't know why :/

Here is my entire configuration : https://www.toptal.com/developers/hastebin/egazenuhoq.json

**Reply by lnu** [reply]
yes but on my side it also returns all the time a value with no decimals. Only in wsl(linux) I get decimals values.  
On windows gopsutil call this method https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-globalmemorystatusex to retrieve the values. I think it returns always a rounded value.  
yes it's explained here: https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/ns-sysinfoapi-memorystatusex

![image](https://user-images.githubusercontent.com/1829553/142850281-c3072d33-8f9e-47d0-aace-dc34a2ec7020.png)

**Reply by Ayfri** [reply]
Updated link to my configuration : https://pastebin.com/x3fRyDCZ
